packages/mui-material/src/Dialog/Dialog.d.ts

```diff
@@ -125,3 +212,5 @@ export interface DialogProps extends StandardProps<ModalProps, 'children'> {
  * - inherits [Modal API](https://mui.com/material-ui/api/modal/)
  */
 export default function Dialog(props: DialogProps): React.JSX.Element;
+
+export interface DialogOwnerState extends DialogProps {}
```

Comment:
```suggestion
export interface DialogOwnerState extends Omit<DialogProps, 'slots' | 'slotProps'> {}
```
to prevent recursive types.

Reply:
fixed https://github.com/mui/material-ui/pull/44792/commits/12c6f4f7fea1fead48d882ec49d61d9bf3599e5b